### DISCUSSION TITLE
sendgrid as a data source provider?

### DISCUSSION THREAD

**Discussion by tx0c** [apache/airflow#24815] [General]
I'm seeing https://github.com/apache/airflow/pull/2727 added Sendgrid EmailOperator, but another question is does Airflow support to synchronize data from Sendgrid as a data source?

some commercial solutions are existing:
1. https://segment.com/docs/connections/sources/catalog/cloud-apps/sendgrid/
2. https://airbyte.com/connections/SendGrid-to-PostgreSQL
3. https://sendgrid.com/blog/pipe-your-sendgrid-data-in-a-data-warehouse/ (was written 2016)
4. https://hevodata.com/learn/sendgrid-to-bigquery/#method2

the need include whatever was written on the sendgrid blog post in 2016:
>> Pipe Your SendGrid Data in a Data Warehouse
> Through Segment, SendGrid customers can:
> - Find the best time of day and day of week to send emails
> - Build a custom funnel measuring email events to in-app conversions
> - Analyze email opens, email link clicks, and email unsubscribes across all your campaigns
> - Get a complete history of how many emails customers receive over time
> - Combine email interactions with in-app behavior to identify high intent prospects and customers likely to churn

but since 2016, during the past 6 years, did someone hit the same need to sync sendgrid data? is there an existing open source solution to synchronize data from Sendgrid?

I have researched this issues board and discussions board, it seemed all threads mentioned sendgrid were using Sendgrid as an EmailOperator,

Thanks,

**Comment by potiuk** [comment]
Not really - but youc could contribute it.

**Reply by tx0c** [reply]
but Airbyte seems to be another competing solution to airflow, and already has it

https://github.com/airbytehq/airbyte/blob/master/docs/integrations/sources/sendgrid.md

**Reply by potiuk** [reply]
Fee free to submit such integration @tx0c. Airlfow is created by > 2100 users like you who contribute usually things they need to have. 
This is a free software and contributing back is the best you can do to pay back the community for it when you use it for free.

I heartily recommend it. You can follow https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst - we even have workshops for new contributors that we run from time to time.

BTW. There is no competition between Aribyte and Airflow. Airflow can actually orchestrate Airbyte job via provider implemented https://airflow.apache.org/docs/apache-airflow-providers-airbyte/stable/index.html so feel free to also use it. We are more than happy for our users to be able to orchestrate Airbyte jobs with Airflow.
